packages/ketcher-react/src/script/ui/views/modal/components/document/Open/Open.tsx

```diff
@@ -113,7 +116,19 @@ const Open: FC<Props> = (props) => {
 
   // @TODO after Recognize is refactored this will not be necessary
   // currently not destructuring onOk with other props so we can pass it with ...rest to Recognize below
-  const { onOk } = rest;
+
+  const ketcher = ketcherProvider.getKetcher();
+  const onOk = (res) => {
+    (
+      dispatch(
+        load(res.structStr, { badHeaderRecover: true, fragment: res.fragment }),
+      ) as unknown as Promise<any>
+    ).catch((error) => {
+      if (ketcher && ketcher.eventBus) {
+        ketcher.eventBus.emit(GLOBAL_ERROR_HANDLER, error.message);
+      }
+    });
+  };
```

Comment:
Maybe we could use` try catch`，which I checked the `load` func already implemented  in the parent component.

Reply:
Apologies for not making myself clear. What I meant was that there's already a try-catch for `load` func in `shared.ts`, so it may not be needed here.

Reply:
OK, got it, thanks @StarlaStarla))